### ISSUE TITLE
Mass withdrawal support

### ISSUE THREAD

**Issue by devWaleed** [kevupton/laravel-coinpayments#14]
Hey, I see no method for mass withdrawal option. I tried to implement by own code by overriding apiCall()  methods but couldn't succeed because apiCall() in Coinpayments returns a single object of receipt model. 

I think if you move new receipt model code to seperate method which can be overriden itself then we can easily implement a mass withdrawal option.

My code:

```
class CoinpaymentsCommandsExtended extends CoinpaymentsCommands
{
    const CREATE_MASS_WITHDRAWAL = 'create_mass_withdrawal';
}

class CoinpaymentsExtended extends Coinpayments
{

    public static function test()
    {

        return CoinpaymentsCommandsExtended::CREATE_MASS_WITHDRAWAL;

    }

    public static function createMassWithdrawal($withdrawalArray)
    {
     // example mass withdrawal array, only for testing purpose
        $req = array(
                'wd[wd1][amount]' => 1.00,
                'wd[wd1][address]' => '1BitcoinAddress',
                'wd[wd1][currency]' => 'BTC',
                'wd[wd2][amount]' => 0.0001,
                'wd[wd2][address]' => '1BitcoinAddress',
                'wd[wd2][currency]' => 'BTC'               
        );

        return self::apiCall(CoinpaymentsCommandsExtended::CREATE_MASS_WITHDRAWAL, $req);

    }
}
```

**Comment by devWaleed** [comment]
```
protected function apiCall ($cmd, $req = [])
    {
        $receipt = parent::apiCall($cmd, $req); <-- can't get pass this because this will return single receipt model, should be array of receipts for mass withdrawal option
        $has_error = $receipt->hasError();
        cp_log([
                'request'  => $receipt->getRequest(),
                'response' => $receipt->getResponse(),
        ], $has_error ? 'API_CALL_ERROR' : 'API_CALL',
                $has_error ? Log::LEVEL_ERROR : Log::LEVEL_ALL
        );

        if ($has_error) {
            throw new CoinPaymentsResponseError($receipt->getError());
        }

        $data = $receipt->toArray();
        if (isset($data['id'])) {
            $data['ref_id'] = $data['id'];
            unset($data['id']);
        }
        switch ($receipt->getCommand()) {
            case CoinpaymentsCommands::CREATE_TRANSACTION:
                return Transaction::create($data);
            case CoinpaymentsCommands::CREATE_WITHDRAWAL:
                return Withdrawal::create($data);
            case CoinpaymentsCommands::CREATE_TRANSFER:
                return Transfer::create($data);
            case CoinpaymentsCommands::GET_CALLBACK_ADDRESS:
                return CallbackAddress::create($data);
            case CoinpaymentsCommandsExtended::CREATE_MASS_WITHDRAWAL:
                return Withdrawal::insert($data);
        }

        return $receipt->getResponse()['result'];
    }
```

**Comment by devWaleed** [comment]
![12345](https://user-images.githubusercontent.com/4759525/38551584-fdd48888-3cd2-11e8-9294-e4963257846e.PNG)

Sending a mass withdrawal command from createMassWIthdrawal method which internally calls the parent Coinpayment apiCall method, returns single receipt like this. If its of any help.

**Comment by kevupton** [comment]
Awesome, thanks. 
I will look into adding this feature. Thanks for sending through all the details as well. this will help a lot 😄

**Comment by kevupton** [comment]
This is now implemented as of version `v1.1.0`

```php
$mass_withdrawal = \Coinpayments::createMassWithdrawal([
    [
        'amount' => $amount1,
        'address' => $address1,
        'currency' => $currency1,
    ],
    [
        'amount' => $amount2,
        'address' => $address2,
        'currency' => $currency2,
    ],
]);
```
